src/renderers/WebGLRenderer.js

```diff
@@ -1524,10 +1522,6 @@ function WebGLRenderer( parameters ) {
 
 				needsProgramChange = true;
 
-			} else if ( materialProperties.vertexAlpha !== vertexAlpha ) {
```

Comment:
I'm not sure why is this needed for...

Reply:
These is required otherwise it won't work if a material is shared between a geometry with vertex colors and without.

Reply:
Some background information: https://github.com/mrdoob/three.js/pull/20975#discussion_r588901717

Reply:
Okay, I follow. Thanks!